bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -247,19 +247,31 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
   }
 
   @Override
-  public ListenableFuture<ImmutableList<SampleRowKeysResponse>> sampleRowKeysAsync(
+  public ListenableFuture<List<SampleRowKeysResponse>> sampleRowKeysAsync(
       SampleRowKeysRequest request) {
-    ClientCall<SampleRowKeysRequest, SampleRowKeysResponse> call =
-        channel.newCall(BigtableServiceGrpc.METHOD_SAMPLE_ROW_KEYS, CallOptions.DEFAULT);
-    CollectingStreamObserver<SampleRowKeysResponse> responseBuffer =
-        new CollectingStreamObserver<>(call);
-    ClientCalls.asyncServerStreamingCall(
-        call,
-        request,
-        responseBuffer);
-    return Futures.transform(
-        responseBuffer.getResponseCompleteFuture(),
-        IMMUTABLE_LIST_FUNCTION);
+    ReadAsync<SampleRowKeysRequest, SampleRowKeysResponse> readAsync =
+        new ReadAsync<SampleRowKeysRequest, SampleRowKeysResponse>() {
+      @Override
+      public ListenableFuture<List<SampleRowKeysResponse>> readAsync(SampleRowKeysRequest request) {
+        ClientCall<SampleRowKeysRequest, SampleRowKeysResponse> call =
+            channel.newCall(BigtableServiceGrpc.METHOD_SAMPLE_ROW_KEYS, CallOptions.DEFAULT);
+        CollectingStreamObserver<SampleRowKeysResponse> responseBuffer =
+            new CollectingStreamObserver<>(call);
+        ClientCalls.asyncServerStreamingCall(
+            call,
+            request,
+            responseBuffer);
+        return Futures.transform(
+            responseBuffer.getResponseCompleteFuture(),
+            IMMUTABLE_LIST_FUNCTION);
+      }};
+
+    if (retryOptions.enableRetries()) {
```

Comment:
Can the 

if (enableretries) {
  ..
 Futures.withFallback(readAsyc.. ...);
} else {
  return readAsyc.readAsyc...
}

be a method that both readSampleRowKeys and readRows use?